src/standalone/Selector/BaseSelector.tsx

```diff
@@ -0,0 +1,244 @@
+		disabled,
+		placeholder,
+		autocompleteId,
+		addNewLabel,
+		onLoad,
+		onAddNew,
+		enableIcons,
+		noOptionsText,
+		loadingText,
+		disableClearable,
+	} = props;
+	const [open, setOpen] = React.useState(false);
+	const actualAddNewLabel =
+		addNewLabel || i18n.t("standalone.selector.add-new");
+	const [selectorOptions, setSelectorOptions] = React.useState(
+		[] as BaseSelectorData[]
+	);
+	const loading = open && selectorOptions.length === 0;
+
+	const delay = (ms: number) => new Promise((res) => setTimeout(res, ms));
+
+	const defaultRenderer = React.useCallback(
+		(data: BaseSelectorData) => (
+			// eslint-disable-next-line @typescript-eslint/ban-ts-comment
+			// @ts-ignore: Typescript complains about the button property being "required"
+			<SelectorSmallListItem>
+				{enableIcons && <SmallListItemIcon>{data.icon}</SmallListItemIcon>}
+				<ListItemText>{data.label}</ListItemText>
+			</SelectorSmallListItem>
+		),
+		[enableIcons]
+	);
+
+	const onChangeHandler = React.useCallback(
+		(data) => {
+			if (
+				data &&
+				"isAddNewButton" in data &&
+				(data as BaseSelectorData).isAddNewButton
+			) {
+				if (onAddNew) onAddNew();
+				return;
+			} else {
+				if (onSelect) onSelect(data);
+			}
+		},
+		[onSelect, onAddNew]
+	);
+
+	const onSearchHandler = React.useCallback(
+		async (query: string) => {
+			const results = await onLoad(query);
+			await delay(2000);
+			if (onAddNew) {
+				results.push({
+					value: "add-new-button",
+					label: actualAddNewLabel,
+					icon: <AddIcon />,
+					isAddNewButton: true,
+				} as BaseSelectorData);
+			}
+			setSelectorOptions(results);
+		},
+		[actualAddNewLabel, onAddNew, onLoad]
+	);
+
+	const setDefaultOptions = React.useCallback(() => {
+		let options = defaultOptions || [];
+		if (onAddNew) {
+			options.push({
+				value: "add-new-button",
+				label: actualAddNewLabel,
+				icon: <AddIcon />,
+				isAddNewButton: true,
+			} as BaseSelectorData);
+		} else {
+			options = options.filter((option) => !option.isAddNewButton);
+		}
+		return options;
+	}, [actualAddNewLabel, defaultOptions, onAddNew]);
+
+	React.useEffect(() => {
+		if (open) {
+			setSelectorOptions(setDefaultOptions);
+		} else {
+			setSelectorOptions([]);
+		}
+	}, [open, defaultOptions, setDefaultOptions]);
+
+	return (
+		<div>
+			<Autocomplete
+				id={autocompleteId}
+				open={open}
+				onOpen={() => {
+					setOpen(true);
+				}}
+				onClose={() => {
+					setOpen(false);
+				}}
+				disableClearable={disableClearable}
+				loading={loading}
+				loadingText={loadingText}
+				placeholder={placeholder}
+				autoComplete
+				disabled={disabled}
+				options={selectorOptions}
+				value={selected}
+				popupIcon={<ExpandMore />}
+				noOptionsText={noOptionsText}
+				getOptionLabel={(option: BaseSelectorData) => option.label}
+				renderOption={(option: BaseSelectorData) => defaultRenderer(option)}
+				getOptionDisabled={(option: BaseSelectorData) =>
+					option.isDisabled as boolean
```

Comment:
instead of casting just write `!!option.isDisabled` or `option.isDisabled ?? false` (this will drop the undefined type, latter allows you to easily set a default value if undefined)